### DISCUSSION TITLE
Compiler

### DISCUSSION THREAD

**Discussion by amirghazi87** [pashmaklang/pashmak#58] [Ideas]
Now pashmak is a interpreter to run files
if we can create a compiler for it for example to python it is easier than other languages

**Comment by mehanalavimajd** [comment]
Not bad idea.

**Comment by parsampsh** [comment]
This is so hard work to compile it to python. but something like **Pyinstaller** is possible. PyInstaller brings the python interpreter binary, and puts the python code in that and packs the output. output is **Python interpreter WITH project's python code**. Maybe possible create something like this to mix the pashmak code and inject it to interpreter binary and generate the output binary. Actually, I'm not literate about this. But we may be able to do that in the future.

**Reply by mehanalavimajd** [reply]
Why we don't do this right now?

**Reply by parsampsh** [reply]
> Why we don't do this right now?

I'm not literate about this :)

**Reply by mehanalavimajd** [reply]
So we can study about this

**Reply by parsampsh** [reply]
> So we can study about this

Sure :) but now we should focus on pashmak self. when a stable release was ready, we can write something like pyinstaller for pashmak.

**Comment by mehanalavimajd** [comment]
But I think we have to compile it to c because it's faster than python

**Reply by parsampsh** [reply]
Interpretive languages ​​are not technically convertible to C/Other languages. Because they have not compiler. they have interpreter. and interpreter is needed to run this. But something like **Injecting code to interpreter binary** is possible(Like pyinstaller). As i said, i'm not literate in this

**Reply by amirghazi87** [reply]
Yes , It is better

**Reply by amirghazi87** [reply]
> Interpretive languages ​​are not technically convertible to C/Other languages. Because they have not compiler. they have interpreter. and interpreter is needed to run this. But something like **Injecting code to interpreter binary** is possible(Like pyinstaller). As i said, i'm not literate in this
I say create new repository and start it to create a compiler to C no in this version
Like two version : compiler and interpreter

**Reply by mehanalavimajd** [reply]
> Interpretive languages ​​are not technically convertible to C/Other languages. Because they have not compiler. they have interpreter. and interpreter is needed to run this. But something like **Injecting code to interpreter binary** is possible(Like pyinstaller). As i said, i'm not literate in this

I think you are right

**Reply by mehanalavimajd** [reply]
> > Interpretive languages ​​are not technically convertible to C/Other languages. Because they have not compiler. they have interpreter. and interpreter is needed to run this. But something like **Injecting code to interpreter binary** is possible(Like pyinstaller). As i said, i'm not literate in this
> > I say create new repository and start it to create a compiler to C no in this version
> > Like two version : compiler and interpreter

Funny and cool idea

**Reply by mehanalavimajd** [reply]
But I think it's hard to make compiler because have to know about c but unfortunately we don't know

**Reply by mehanalavimajd** [reply]
> > Interpretive languages ​​are not technically convertible to C/Other languages. Because they have not compiler. they have interpreter. and interpreter is needed to run this. But something like **Injecting code to interpreter binary** is possible(Like pyinstaller). As i said, i'm not literate in this
> > I say create new repository and start it to create a compiler to C no in this version
> > Like two version : compiler and interpreter

What do you think about this @parsampsh

**Reply by parsampsh** [reply]
So i haven't any idea about this. because this needs re-write pashmak and change base structures. already pashamk is very depended to python.

**Reply by parsampsh** [reply]
Also as you said, this needs to know low level programming with C/C++ and i don't know :). also this project is depended to python. lot of features in backend are handled by python. so this cannot be splited from python.

**Comment by parsampsh** [comment]
Guys. totally something like pyinstaller is good idea. we can write a program to mix our pashmak code. then inject it to pashmak interpreter's binary file. then generate the result as a binary executable file. this is possible. but currently i'm not literate in this. i will read about this in the future. but now, we should focus on pashmak self and make a stable release. then we can work on the side projects.

**Comment by parsampsh** [comment]
Hi guys. I wrote a simple **Mixer**  for pashmak. this can mix multiple file project files to ONE file.
https://github.com/pashmaklang/pashmiler

**Comment by adabadadudu** [comment]
but we can create a source-to-source transcompiler using any language to an faster language
